### PR TITLE
Add support for a {:struct, struct_name} type specifier

### PR DESCRIPTION
**Pull request by aselder**

### CODE REVIEW

File: lib/nimble_options/docs.ex [hunk 1]
```diff
@@ -127,6 +127,9 @@ defmodule NimbleOptions.Docs do
   defp get_raw_type_str({:non_empty_keyword_list, _keys}), do: "non-empty `t:keyword/0`"
   defp get_raw_type_str(:map), do: "`t:map/0`"
 
+  defp get_raw_type_str({:struct, struct_type}),
+    do: "struct of type #{nice_struct_name(struct_type)}"
```
Comment (whatyouhide) [inline lib/nimble_options/docs.ex:131]:
```suggestion
    do: "struct of type #{inspect(struct_type)}"
```

File: lib/nimble_options/docs.ex [hunk 2]
```diff
@@ -259,4 +265,10 @@ defmodule NimbleOptions.Docs do
     |> Enum.reverse()
     |> Enum.reduce(&quote(do: unquote(&1) | unquote(&2)))
   end
+
+  defp nice_struct_name(struct_type) do
+    struct_type
+    |> Module.split()
+    |> Enum.join(".")
+  end
```
Comment (whatyouhide) [inline lib/nimble_options/docs.ex:273]: You don't need this, you can use `inspect/1` as suggested above 🙃 

```suggestion
```

**Review by whatyouhide — APPROVED** [review]

**Comment by whatyouhide** [comment]
@aselder this PR is failing on Elixir 1.9 (still supported by this library) because `is_struct/2` was introduced later on. Can you move the check to the body of the function, and use something like `Map.has_key?(struct, :__struct__) and struct.__struct__ == mod`?

**Comment by josevalim** [comment]
We can pattern match as well:

    match?(%^mod{}, value)

**Comment by aselder** [comment]
Any idea what a rough timeline for getting this released is? Trying decide if I launch a feature with out and go back and update later, or wait for this to make it out. Thanks!

**Comment by aselder** [comment]
@whatyouhide Thanks Andrea

**Pull request merged**